mcp/src/util/addMessageLinks.ts

```diff
@@ -1,14 +1,51 @@
+import { log } from '../shared/boilerplate/src/logger.js';
 import { Message } from '../types.js';
 import { convertTimestampToTs } from './formatTs.js';
 
-const SLACK_DOMAIN = process.env.SLACK_DOMAIN;
+const SLACK_BOT_TOKEN = process.env.SLACK_BOT_TOKEN;
+
+let workspaceBaseUrl: string | null;
+
+export const initWorkspaceBaseUrl = async (): Promise<void> => {
+  try {
+    const response = await fetch('https://slack.com/api/auth.test', {
+      method: 'POST',
+      headers: {
+        Authorization: `Bearer ${SLACK_BOT_TOKEN}`,
+        'Content-Type': 'application/json',
+      },
+    });
+
+    const data = await response.json();
+
+    if (!data.ok) {
+      log.warn(
+        'An error occurred while getting the Slack url base. Will not return permalinks.',
+        { data },
+      );
```

Comment:
Could probably consolidate the error logging in the catch below.

```suggestion
      throw new Error('auth.test request failed');
```